### PR TITLE
Add initial UI binary and frontend boilerplate

### PR DESCRIPTION
**Pull request by jpellizzari**
Adds the initial UI wiring and boilerplate:
Screenshot from 2021-05-19 15-56-18

This PR doesn't make any decisions about the API architecture or style as that is still TBD.

A fast follow from this will be the initial styled UI along with and implementation of the `.Add()` method

### CODE REVIEW

**Review by luizbafilho — APPROVED** [review]

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -17,8 +17,28 @@ jobs:
         go-version: 1.16.x
     - name: Checkout code
       uses: actions/checkout@v2
+    - name: Cache node modules
+      uses: actions/cache@v2
```
Comment (josecordaz) [inline .github/workflows/test.yml:21]: great addition

**Review by josecordaz — COMMENTED** [review]
Got this error I probably did something wrong. Could you please confirm ? @jpellizzari  🙂 
image

**Comment by jpellizzari** [comment]
> Got this error I probably did something wrong. Could you please confirm ? @jpellizzari 🙂
> image

Should have added this to the README: `make cmd/ui/dist/index.html`

**Review by josecordaz — APPROVED** [review]
LGTM

**Comment by jpellizzari** [comment]
After a rebase, this branch inherited some failing checks from `main`. Merging anyway after our discussion this morning about the state of the build checks.

**Pull request merged**